### PR TITLE
BRMS-34 adding metrics class and unit tests

### PR DESCRIPTION
**Pull request by adamconder**
- Adding metrics class and methods to time and status count connector.

### CODE REVIEW

File: app/uk/gov/hmrc/brm/metrics/Metrics.scala [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+object Metrics extends Metrics {
+
+  val timer = (name: String) => MetricsRegistry.defaultRegistry.timer(name)
+  val counter = (name: String) => MetricsRegistry.defaultRegistry.counter(name)
+
+  Logger.info("[Metrics][constructor] Preloading metrics keys")
+
+  // register metrics with timer and counter
+  Seq(
```
Comment (chris-ianson) [inline app/uk/gov/hmrc/brm/metrics/Metrics.scala]: @adamconder @manishsw 

Do we need to think about the key format/names? Do the keys below relate to the proxy response or the birth-registration-matching response to the consuming service?

Comment (adamconder) [reply app/uk/gov/hmrc/brm/metrics/Metrics.scala]: @chris-ianson The keys below we're the responses from the proxy. We can add additional keys if necessary for the matching responses.

**Comment by adamconder** [comment]
@chris-ianson I've updated the PR with additional test cases and now works across all connectors.

File: project/MicroServiceBuild.scala [hunk 2]
```diff
@@ -27,10 +27,10 @@ private object AppDependencies {
   private val domainVersion = "3.7.0"
   private val hmrcTestVersion = "1.6.0"
   private val mockito = "1.9.5"
-  private val metricsPlayVersion = "0.2.1"
+  private val metricsPlayVersion = "2.3.0_0.2.1"
+  private val metricsGraphite = "3.0.2"
```
Comment (chris-ianson) [inline project/MicroServiceBuild.scala]: Remove metricsGraphite

File: project/MicroServiceBuild.scala [hunk 3]
```diff
@@ -39,7 +39,8 @@ private object AppDependencies {
     "uk.gov.hmrc" %% "play-config" % playConfigVersion,
     "uk.gov.hmrc" %% "play-json-logger" % playJsonLoggerVersion,
     "uk.gov.hmrc" %% "domain" % domainVersion,
-    "com.kenshoo" %% "metrics-play" % metricsPlayVersion
+    "com.kenshoo" %% "metrics-play" % metricsPlayVersion,
+    "com.codahale.metrics" % "metrics-graphite" % metricsGraphite
```
Comment (chris-ianson) [inline project/MicroServiceBuild.scala]: Remove graphite

File: test/uk/gov/hmrc/brm/metrics/MetricsSpec/MetricsSpec.scala [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+
+    "have a 400 status count for gro-ni" in {
+      val metrics = GRONIMetrics
+      metrics.connectorStatus(400)
+      MetricsRegistry.defaultRegistry.getCounters.get("gro-ni-connector-status-400").getCount shouldBe 1
+    }
+
+    "have a 404 status count for proxy" in {
+      val metrics = GRONIMetrics
+      metrics.connectorStatus(404)
```
Comment (chris-ianson) [inline test/uk/gov/hmrc/brm/metrics/MetricsSpec/MetricsSpec.scala:172]: Ah, so we are a little more specific with error codes. Do you want us to log errors outside of 200, 400 and 500 (if applicable)

**Review by chris-ianson — CHANGES_REQUESTED** [review]
Looks great, just a couple of comments

File: test/uk/gov/hmrc/brm/metrics/MetricsSpec/MetricsSpec.scala (hunk 4 above)
Comment (adamconder) [reply test/uk/gov/hmrc/brm/metrics/MetricsSpec/MetricsSpec.scala:172]: @chris-ianson It would be useful if you log exceptions outside of 400 etc.

Comment (adamconder) [reply test/uk/gov/hmrc/brm/metrics/MetricsSpec/MetricsSpec.scala:172]: Updated these :)

**Review by chris-ianson — APPROVED** [review]

**Pull request merged**